app/src/ui/app.tsx

```diff
@@ -746,12 +746,12 @@ export class App extends React.Component<IAppProps, IAppState> {
         repository,
       })
     } else {
-      this.props.dispatcher.removeRepositories([repository])
+      this.props.dispatcher.removeRepositories([repository], false)
     }
   }
 
   private onConfirmRepoRemoval = (repository: Repository) => {
-    this.props.dispatcher.removeRepositories([repository])
+    this.props.dispatcher.removeRepositories([repository], false)
```

Comment:
I would like to use `state.includeMoveToTrash` here but I'm struggling to extract it :/

Reply:
Check out how we're doing the Discard Changes dialog.

For the application state:

 - introduce some app state that represents the dialog behaviour

https://github.com/desktop/desktop/blob/88082bc0e75c99af4a3afc3f742b48b2c3612332/app/src/lib/app-state.ts#L161-L162

 - set some default state for this value in the App Store

https://github.com/desktop/desktop/blob/88082bc0e75c99af4a3afc3f742b48b2c3612332/app/src/lib/stores/app-store.ts#L241

https://github.com/desktop/desktop/blob/88082bc0e75c99af4a3afc3f742b48b2c3612332/app/src/lib/stores/app-store.ts#L585

For where you need to handle the action:

 - pass the application state down through props

https://github.com/desktop/desktop/blob/88082bc0e75c99af4a3afc3f742b48b2c3612332/app/src/ui/changes/sidebar.tsx#L49

 - in the event handler, react depending on the current state

https://github.com/desktop/desktop/blob/master/app/src/ui/changes/sidebar.tsx#L158-L168

And for the dialog itself:

 - initial state for the dialog is derived from props (that comes from app state)

https://github.com/desktop/desktop/blob/0ce3de645441bb4a30506636210edbaa6975a970/app/src/ui/discard-changes/discard-changes-dialog.tsx#L52-L55

 - when submitting, the action is performed and the state is updated through the dispatcher

https://github.com/desktop/desktop/blob/88082bc0e75c99af4a3afc3f742b48b2c3612332/app/src/ui/discard-changes/discard-changes-dialog.tsx#L136-L146